Analysis/TauFinder/src/TaJetClustering.cc

```diff
@@ -183,8 +184,13 @@ bool compareEnergy( ReconstructedParticle * const &p1, ReconstructedParticle * c
 void TaJetClustering::processEvent( LCEvent * evt ) { 
 
   // obtain Reconstructed Particles
-  LCCollection* col = evt->getCollection( _pfoCollectionName );
-  if( !col ){ /*cout << "Failed to obtain PFO collection!" << endl;*/ return; }
+  LCCollection* col{nullptr};
+  try {
+    col = evt->getCollection(_pfoCollectionName);
+  } catch (lcio::DataNotAvailableException& e) {
+    streamlog_out(WARNING) << "No PFO collection (" << _pfoCollectionName << ") found" << std::endl;
```

Comment:
I would not print this as a WARNING if this is expected in some physics events rather use DEBUG5 or so.
Otherwise the logfiles will be swamped with warnings that are none, really